astropy/time/tests/test_update_leap_seconds.py

```diff
@@ -58,28 +59,27 @@ def test_auto_update_bad_file(self):
             update_leap_seconds(["nonsense"])
 
     def test_auto_update_corrupt_file(self, tmp_path):
-        bad_file = str(tmp_path / "no_expiration")
-        with open(iers.IERS_LEAP_SECOND_FILE) as fh:
-            lines = fh.readlines()
-        with open(bad_file, "w") as fh:
-            fh.write("\n".join([line for line in lines if not line.startswith("#")]))
+        bad_file = tmp_path / "no_expiration"
+        lines = Path(iers.IERS_LEAP_SECOND_FILE).read_text().splitlines()
+        bad_file.write_text(
+            "\n".join(line for line in lines if not line.startswith("#"))
+        )
 
         with pytest.warns(AstropyWarning, match="ValueError.*did not find expiration"):
-            update_leap_seconds([bad_file])
+            # FIXME: update_leap_seconds should be able to handle Path objects
```

Comment:
If this is not an issue already, maybe we should open a GitHub issue and link that here?

Reply:
yes. I just took a very quick look and it's indeed not trivial to fix, so an issue seems appropriate. I can open it tomorrow (unless you want to beat me to it)

Reply:
You can probably provide better details. I can wait. Thanks! 😺

Reply:
done in #16929 (and linked in a comment here)

Reply:
moving this PR to draft so #16931 can be prioritized. This comment can be removed once  #16931 in merged.

Reply:
The linked PR is merged. FYI.

Reply:
thanks for reminding me ! rebased, and soon-to-be undrafted